packages/flutter_tools/lib/src/commands/analyze_base.dart

```diff
@@ -103,6 +113,41 @@ abstract class AnalyzeBase {
 
     return dartDocMessage;
   }
+
+  /// Print an analysis summary.
+  static String generateErrorsMessage({
+    @required int issueCount,
+    int issueDiff,
+    int files,
+    @required String seconds,
+    int undocumentedMembers = 0,
+    String dartDocMessage = '',
+  }) {
+    String errorsMessage;
+    final StringBuffer issuesMessage = StringBuffer(issueCount > 0
+      ? '$issueCount ${pluralize('issue', issueCount)} found.'
+      : 'No issues found!');
+
+    if (issueDiff != null) {
+      if (issueDiff > 0) {
+        issuesMessage.write(' ($issueDiff new)');
+      } else if (issueDiff < 0) {
+        issuesMessage.write(' (${-issueDiff} fixed)');
+      }
+    }
+
+    String filesMessage = '';
+    if (files != null) {
+      filesMessage = ' • analyzed $files ${pluralize('file', files)}';
+    }
+
+    if (undocumentedMembers > 0) {
+      errorsMessage = '$issuesMessage$filesMessage (ran in ${seconds}s; $dartDocMessage)';
```

Comment:
Since you're refactoring this - could you document the reason these two messages are almost the same? Or make it more obvious by other means (moving the rest of the message generation here?)

Reply:
I add dartdoc comments